lib/chef/knife/bmcs_server_create.rb

```diff
@@ -170,6 +182,36 @@ def run
         show_value('Node Name', config[:chef_node_name])
       end
 
+      # Return CreateVnicDetails instance
+      def make_vnic_details
+        vnic_details = OracleBMC::Core::Models::CreateVnicDetails.new
+        vnic_details.assign_public_ip = bool_arg(config[:assign_public_ip]) if config[:assign_public_ip]
+        vnic_details.private_ip = config[:private_ip] if config[:private_ip]
+        vnic_details.subnet_id = config[:subnet_id]
+        vnic_details
+      end
+
+      def get_bootstrap_ip(vnic)
+        return vnic.private_ip if config[:use_private_ip]
+        bool_arg(config[:assign_public_ip]) == false ? vnic.private_ip : vnic.public_ip
```

Comment:
Yep, just what I was thinking. Basically, we don't want to try to infer when the public IP address is available when we already have that info from the service.